R/modules.R

```diff
@@ -321,11 +321,263 @@ modules <- function(..., label = "root") {
 # printing methods ----
 
 #' @rdname teal_modules
+#' @param is_last (`logical(1)`) Whether this is the last item in its parent's children list.
+#'   Affects the tree branch character used (L- vs |-)
+#' @param parent_prefix (`character(1)`) The prefix inherited from parent nodes,
+#'   used to maintain the tree structure in nested levels
+#' @param is_root (`logical(1)`) Whether this is the root node of the tree. Only used in
+#'   format.teal_modules(). Determines whether to show "TEAL ROOT" header
+#' @param what (`character`) Specifies which metadata to display.
+#'   Possible values: "datasets", "properties", "ui_args", "server_args", "transformers"
+#' @examples
+#' mod <- module(
+#'   label = "My Custom Module",
+#'   server = function(id, data, ...) {},
+#'   ui = function(id, ...) {},
+#'   datanames = c("ADSL", "ADTTE"),
+#'   transformers = list(),
+#'   ui_args = list(a = 1, b = "b"),
+#'   server_args = list(x = 5, y = list(p = 1))
+#' )
+#' cat(format(mod))
 #' @export
-format.teal_module <- function(x, indent = 0, ...) {
-  paste0(paste(rep(" ", indent), collapse = ""), "+ ", x$label, "\n", collapse = "")
+format.teal_module <- function(
+    x, indent = 0, is_last = FALSE, parent_prefix = "",
+    what = c("datasets", "properties", "ui_args", "server_args", "transformers"), ...) {
+  empty_text <- "NULL"
```

Comment:
Hey @vedhav I thought about regular NULL (not a character) and empty spaces in the final print. Now we get a NULL sign.

Reply:
I don't get what you mean. Can you please explain what you mean by `NULL` and empty spaces?

Reply:
Instead of 

```r
empty_text <- "NULL"
```

try 

```r
empty_text <- NULL
```

Reply:
And those are empty spaces that are in the output 
<img width="391" alt="image" src="https://github.com/user-attachments/assets/151f9f2d-4f79-4bd7-94e8-001af7cdc69e">

Reply:
For me the empty spaces only come when I have `NULL` and not `"NULL"`

Reply:
Yes, that was my idea. Let's change `"NULL"` to `""`